webui/portal/src/main/java/org/exoplatform/portal/application/localization/DefaultLocalePolicyService.java

```diff
@@ -47,6 +64,48 @@
  * @author <a href="mailto:mstrukel@redhat.com">Marko Strukelj</a>
  */
 public class DefaultLocalePolicyService implements LocalePolicy, Startable {
```

Comment:
In this service we implemented two methods one for HTTp context ( using the LocaleContextInfo) and the other for Services context ( no http context ) witch based on the Portal config.